Frontend/library/src/Util/CoordinateConverter.ts

```diff
@@ -1,218 +1,73 @@
 // Copyright Epic Games, Inc. All Rights Reserved.
 
-import { Logger } from '@epicgames-ps/lib-pixelstreamingcommon-ue5.5';
-import { VideoPlayer } from '../VideoPlayer/VideoPlayer';
+export interface RectSize {
+    width: number;
+    height: number;
+}
+
+export interface NormalizedQuantizedUnsignedCoord {
+    inRange: boolean;
+    x: number;
+    y: number;
+}
+
+export interface UnquantizedDenormalizedUnsignedCoord {
+    x: number;
+    y: number;
+}
+
+export interface NormalizedQuantizedSignedCoord {
+    x: number;
+    y: number;
+}
 
 /**
  * Converts coordinates from element relative coordinates to values normalized within the value range of a short (and back again)
  */
 export class CoordinateConverter {
-    videoElementProvider: VideoPlayer;
-    videoElementParent: HTMLElement;
-    videoElement: HTMLVideoElement;
+    playerSize: RectSize;
     ratio: number;
-
-    normalizeAndQuantizeUnsignedFunc: (x: number, y: number) => NormalizedQuantizedUnsignedCoord;
-    normalizeAndQuantizeSignedFunc: (x: number, y: number) => NormalizedQuantizedSignedCoord;
-    denormalizeAndUnquantizeUnsignedFunc: (x: number, y: number) => UnquantizedDenormalizedUnsignedCoord;
-
-    /**
-     * @param videoElementProvider - the div element that the video player will be injected into
-     */
-    constructor(videoElementProvider: VideoPlayer) {
-        this.videoElementProvider = videoElementProvider;
-        this.normalizeAndQuantizeUnsignedFunc = () => {
-            throw new Error('Normalize and quantize unsigned, method not implemented.');
-        };
-        this.normalizeAndQuantizeSignedFunc = () => {
-            throw new Error('Normalize and unquantize signed, method not implemented.');
-        };
-        this.denormalizeAndUnquantizeUnsignedFunc = () => {
-            throw new Error('Denormalize and unquantize unsigned, method not implemented.');
-        };
+    playerIsLarger: boolean;
+
+    // we dont use a constructor here because the object is created and passed around to various locations
+    // possibly before this method is called.
+    configure(playerSize: RectSize, videoSize: RectSize) {
+        const playerAspectRatio = playerSize.height / playerSize.width;
+        const videoAspectRatio = videoSize.height / videoSize.width;
+        this.playerIsLarger = playerAspectRatio > videoAspectRatio;
+        this.playerSize = playerSize;
```

Comment:
Caution about passing in a playerSize and videoSize like this, I think they can change as the page is resized, hence why the element was used before.

Reply:
Configure is called from an onresize event, Thats why i called it `configure` instead of `init` or `setup` or similar.